www/nginx/src/opnsense/mvc/app/models/OPNsense/Base/Constraints/NgxUniqueDefaultServerConstraint.php

```diff
@@ -0,0 +1,139 @@
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* 1. Redistributions of source code must retain the above copyright notice,
+*    this list of conditions and the following disclaimer.
+*
+* 2. Redistributions in binary form must reproduce the above copyright
+*    notice, this list of conditions and the following disclaimer in the
+*    documentation and/or other materials provided with the distribution.
+*
+* THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+* INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+* AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+* OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+namespace OPNsense\Base\Constraints;
+
+use Phalcon\Messages\Message;
+
+/**
+ * a very specific nginx check - not reusable
+ *
+ * Class NgxUniqueDefaultServerConstraint
+ * @package OPNsense\Nginx\Constraints
+ */
+class NgxUniqueDefaultServerConstraint extends BaseConstraint
+{
+    public function validate(\Phalcon\Validation $validator, $attribute): bool
+    {
+        $node = $this->getOption('node');
+        if ($node) {
+			$httpServerNode = $node->getParentNode();
+			$defaultServerNode = $httpServerNode->getChild("default_server");
+            if (!$this->isEmpty($defaultServerNode))
+            {
+				$myUUID = $httpServerNode->getAttribute("uuid");
+				$myListenHTTPAddress = $httpServerNode->getChild("listen_http_address");
+				
+				$httpServersNode = $httpServerNode->getParentNode();
+				
+				$httpServers = $httpServersNode->getChildren();
+
+				$msg = "";				
+				foreach ($httpServers as $httpServer)
+				{
+					$uuid = $httpServer->getAttribute("uuid");
+					if ($uuid != $myUUID)
+					{
+						$defaultServerNode = $httpServer->getChild("default_server");
+						if (!$this->isEmpty($defaultServerNode))
+						{
+							$listenHTTPAddressNode = $httpServer->getChild("listen_http_address");
+							if ($this->compareListenAddresses($myListenHTTPAddress, $listenHTTPAddressNode, $msg))
+							{
+								$msg = "There can only be one Default Server on each listening address: " . $msg . " conflict.";
+								$validator->appendMessage(new Message(
+									gettext($msg),
+									$attribute
+								));
+							}
+						}
+					}
+					
+				}
+
+				
+			}
+			
+
+        }
+        return true;
+    }
+    
+    private function compareListenAddresses($as, $bs, &$msg)
```

Comment:
if you know the type, it would be cool if you could type the method. In that case, it would be a bool return type.

To add it, you would have to add ": bool" to this line.

If you do this, the interpreter checks that it will return a bool or throw an error. The advantage is, that IDEs know the type as well and the syntax checks work better.

Reply:
done